### DISCUSSION TITLE
[react] HTMLAttributes does not accept null

### DISCUSSION THREAD

**Discussion by ntkoopman** [DefinitelyTyped/DefinitelyTyped#61538] [Issues with a @types package]
All properties on `HTMLAttributes` only allow `undefined` but not `null`, but React will accept `null` without issue. This is very annoying when working with data sources that return `null`, because you need to cast using `?? undefined` everywhere.

A bad example:
```tsx
const title: string | null = null;
<meta name="title" content={title}/>; // error TS2322 not assignable, but would work at runtime
<meta name="title" content={title ?? undefined}/>; // works, but gets old very quickly
```

This is the same issue as https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/55129. I'm wondering if there is any reason `null` is not included?

[a comment by an automated account is not shown]

**Comment by Jessidhia** [comment]
I checked the code to see if switching between `null` and `undefined` could be a reconciliation hazard but [ReactDOM seems to handle it](https://github.com/facebook/react/blob/9fcaf88d58cfd942e2fdd303ae8291dbf4828969/packages/react-dom/src/client/ReactDOMComponent.js#L694-L700) (note the use of `==`), so there is not a fundamental technical reason to disallow it. There may still be a code quality or intended usage reason, though.

**Reply by ntkoopman** [reply]
To add to this, in Flow everything is typed as `any`: https://github.com/facebook/flow/blob/main/lib/react-dom.js#L679
